### PR TITLE
Created the NFT component for single NFT allowance

### LINKED ISSUE #15702: Create the NFT component for single NFT allowance
## Resources
[Figma Design file](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=2007%3A14482)

## Acceptance Criteria
1. Icon displayed is the NFT collection icon
2. Copy in bold is the NFT collection name
3. Secondary copy is specific token ID for the NFT that is being request allowance for 
4. Clicking on "View" opens up the "View NFTs" modal: https://github.com/MetaMask/metamask-extension/issues/15751
Screen Shot 2022-08-24 at 19 54 10

### PR DESCRIPTION
**Pull request by adnansahovic**
## Explanation

Created the NFT component for single NFT allowance. This component can't yet be added to the app. Instead, I added a storybook that displays this component.

1. Icon displayed is the NFT collection icon
2. Copy in bold is the NFT collection name
3. Secondary copy is specific token ID for the NFT that is being request allowance for
4. Clicking on "View" opens up the "View NFTs" modal: https://github.com/MetaMask/metamask-extension/issues/15751

## More Information

* Fixes #15702

## Screenshots/Screencaps

Screenshot 2022-09-14 at 10 02 04

## Manual Testing Steps

This component can be tested by running storybook (yarn storybook) and then go to the COMPONENTS tab and open in UI/NftInfo.

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

File: ui/components/ui/nft-info/nft-info.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            color={COLORS.TEXT_ALTERNATIVE}
+          >
+            {t('tokenId')} #{tokenId}
+          </Typography>
+        </Box>
+        <Box
+          textAlign={TEXT_ALIGN.END}
+          marginTop={4}
+          marginRight={4}
+          className="nft-info__content__view"
```
Comment (brad-decker) [inline ui/components/ui/nft-info/nft-info.js:55]: BEM class syntax doesn't allow for nesting blocks this way, if you need a block within another block then it should be named like `nft-info__content-view`

File: ui/components/ui/nft-info/index.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.nft-info {
+  &__content {
+    width: 100%;
+
+    &__view {
+      flex-grow: 1;
```
Comment (brad-decker) [inline ui/components/ui/nft-info/index.scss:6]: I'm mobile at the moment, so this is just a question, do we not currently support flexGrow as a property of the Box component?

File: ui/components/ui/nft-info/nft-info.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        >
+          <button className="nft-info__content__view__button" type="link">
```
Comment (brad-decker) [inline ui/components/ui/nft-info/nft-info.js:57]: This should be our Button design component.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/components/ui/nft-info/index.scss (hunk 2 above)
Comment (adnansahovic) [reply ui/components/ui/nft-info/index.scss:6]: Hi @brad-decker, flexGrow is currently not supported

File: ui/components/ui/nft-info/nft-info.js (hunk 1 above)
Comment (adnansahovic) [reply ui/components/ui/nft-info/nft-info.js]: Hi @brad-decker , You can see changes [here](https://github.com/MetaMask/metamask-extension/pull/15825/commits/a8832a59e477108990dfbe5146909100bebb7d2e#diff-8507ca3fcec29b0d76237eefb900b4688d880b82c315e4809ed9e84fc644fb41)

File: ui/components/ui/nft-info/nft-info.js (hunk 3 above)
Comment (adnansahovic) [reply ui/components/ui/nft-info/nft-info.js]: Hi @brad-decker, Done [here](https://github.com/MetaMask/metamask-extension/pull/15825/commits/a8832a59e477108990dfbe5146909100bebb7d2e#diff-8507ca3fcec29b0d76237eefb900b4688d880b82c315e4809ed9e84fc644fb41)

**Review by brad-decker — DISMISSED** [review]

**Review by brad-decker — APPROVED** [review]
LGTM

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Pull request merged**